sqlmesh/core/table_diff.py

```diff
@@ -0,0 +1,195 @@
+from __future__ import annotations
+
+import typing as t
+
+import pandas as pd
+from sqlglot import exp
+from sqlglot.dialects.dialect import DialectType
+
+from sqlmesh.utils.pydantic import PydanticModel
+
+if t.TYPE_CHECKING:
+    from sqlmesh.core._typing import TableName
+    from sqlmesh.core.engine_adapter import EngineAdapter
+
+
+class SchemaDiff(PydanticModel, frozen=True):
```

Comment:
Please note that we already have a concept of a schema diff. Perhaps worth disambiguating

Reply:
although there's only a SchemaDiffer not a schema diff, what do you want to rename it to?